drivers/shared/validators/validators.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package validators
+
+import (
+	"fmt"
+	"os/user"
+	"strconv"
+	"strings"
+)
+
+// IDRange defines a range of uids or gids (to eventually restrict)
+type IDRange struct {
+	Lower uint64 `codec:"from"`
+	Upper uint64 `codec:"to"`
+}
+
+// IDRangeValid is used to ensure that the configuration for ID ranges is valid.
+func IDRangeValid(rangeType string, deniedRanges string) error {
+	_, err := parseRanges(deniedRanges)
+	if err != nil {
+		return fmt.Errorf("invalid %s value %q: %v", rangeType, deniedRanges, err)
+	}
+
+	return nil
+}
+
+type userLookupFn func(string) (*user.User, error)
+
+// UserInRange is used when running a task to ensure the
+// given user is in the ID range defined in the task config
+func UserInRange(userLookupFn userLookupFn, usernameToLookup string, deniedHostUIDs, deniedHostGIDs string) error {
+
+	// look up user on host given username
+
+	u, err := userLookupFn(usernameToLookup)
+	if err != nil {
+		return fmt.Errorf("failed to identify user %q: %v", usernameToLookup, err)
+	}
+	uid, err := strconv.ParseUint(u.Uid, 10, 32)
+	if err != nil {
+		return fmt.Errorf("unable to convert userid %s to integer", u.Uid)
+	}
+
+	// check uids
+
+	uidRanges, err := parseRanges(deniedHostUIDs)
```

Comment:
I'm pretty sure we already do this earlier when we read the plugin's configuration? So we should be able to pass the already-parsed `IDRange` object into this function so that we only do it once at plugin startup, instead of on every task start.

Reply:
We do do this twice, but it is happening on different machines. I'm pretty sure the initial validate is happening by the server when the job is submitted, and then this bit is happening when the task is starting on the client (and needs to happen there cause we only have the user info then). We could parse the jobspec and then save the parsed IdRanges in state somewhere and pass those down to the clients, but I figured that was not worth it versus just running the parsing again. Thoughts?

Reply:
The server can't validate anything about the driver (either the driver's config or anything in the `task.config` block). Generally speaking, nothing in the driver code ever happens on the server -- not being able to validate driver config on the server is a long-standing user complaint, in fact.

Reply:
Oh blegh, sorry I'm being dumb! - Lemme see if I can thread this thru

Reply:
Oh man, I was being dense and (briefly) thinking this was on job submit, but its really on driver config load...